plugin/storage/es/spanstore/reader.go

```diff
@@ -128,23 +122,23 @@ func NewSpanReader(p SpanReaderParams) *SpanReader {
 	if p.UseReadWriteAliases {
 		maxSpanAge = rolloverMaxSpanAge
 	}
+
+	p.SpanIndex.Prefix = p.SpanIndex.FullIndexName(spanIndexBaseName)
+	p.ServiceIndex.Prefix = p.ServiceIndex.FullIndexName(serviceIndexBaseName)
```

Comment:
@yurishkuro regarding the above, it looks like we have both span index and service index when constructing the span reader here

Reply:
Reader doesn't matter as much because service index is only used for GetServices call, which would never be dispatched to the archive storage. But the archive writer could try to write to both spans and services indices. I just checked and doesn't seem like that would happen: https://github.com/jaegertracing/jaeger/blob/f411b3c49ccf5fb54abe3a024e772db28e4fc5b2/plugin/storage/es/spanstore/writer.go#L108

Here the empty string is returned for the services index name, and with blank string service name won't be written: https://github.com/jaegertracing/jaeger/blob/f411b3c49ccf5fb54abe3a024e772db28e4fc5b2/plugin/storage/es/spanstore/writer.go#L131

So I think the config in the original PR would still work, but one other thing I want to check is how the `isArchive` flag is being passed when creating ES Storage. In case of v1 it happens because v1 calls `factory.CreateArchiveWriter`, but in v2 all storages are initialized the same way so I don't think we actually call that function, we just call the main `CreateWriter`.

Reply:
Booked [#6065](https://github.com/jaegertracing/jaeger/issues/6065)